t/unit/utils/test_collections.py

```diff
@@ -448,3 +449,16 @@ def test_pop_empty_no_default(self):
 
     def test_repr(self):
         assert repr(Messagebuffer(10, [1, 2, 3]))
+
+
+class test_ChainMap:
+
+    def test_observers_not_shared(self):
+        a = ChainMap()
+        b = ChainMap()
+        callback = Mock()
+        a.bind_to(callback)
+        b.update(x=1)
+        callback.assert_not_called()
```

Comment:
can we lean to pytest like syntax?

Reply:
is there an alternative to `assert_not_called` and `assert_called_once_with` in pytest?
I see they are used extensively in celery unit tests.

Reply:
ugh sorry my mistake!

Reply:
do you need anything else from me?

Reply:
is it possible to increase test coverage and if possible to add integration tests as well?

Reply:
It's weird that I reduced coverage by this commit - It's just a bugfix with a new unit test that checks it actually fixed the bug (I didn't add any new "logic" to the code itself).
It also feels like a low-level bug that doesn't suit an integration test.
WDYT?

Reply:
codecov sometimes do mistakes :D